File: collatzConjecture/collatz.cpp
#include <iostream>
using namespace std;
int main(){
    cout<<"Enter a positive number: \n";
    int n;
    cin>>n;
    if (n>0)
    {
        do
        {
            cout<<n<<endl;
            if(n%2==0) n/=2;
            else if(n%2!=0) n=n*3+1;
            if(n==1) cout<<n;
        } while (n!=1);
        
    }else cout<<"Not a positive number. Exiting.";
    
}

File: collatzConjecture/collatz.py
n = int(input('Enter a positive number: \n'))
while 1:
    print(n)
    if (n<0):
        print('Not a positive number. Exiting.')
        break
    elif n==1:
        break
    if n%2==0:
        n /=2
    elif n%2!=0:
        n = 3*n+1